File: daizhang-frontend/src/types/subject.ts
export interface SubjectVO {
  id: number
  accountSetId: number
  subjectCode: string
  subjectName: string
  category: string
  parentId: number
  level: number
  subjectType: number
  balanceDirection: number
  auxiliaryAccounting: number
  isCash: number
  isBank: number
  isCurrent: number
  status: number
  createTime: string
  children?: SubjectVO[]
}

export interface SubjectCreateRequest {
  accountSetId: number
  subjectCode: string
  subjectName: string
  category?: string
  parentId?: number
  level?: number
  subjectType?: number
  balanceDirection?: number
  auxiliaryAccounting?: number
  isCash?: number
  isBank?: number
  isCurrent?: number
}

export interface SubjectUpdateRequest {
  subjectName?: string
  category?: string
  subjectType?: number
  balanceDirection?: number
  auxiliaryAccounting?: number
  isCash?: number
  isBank?: number
  isCurrent?: number
  status?: number
}

File: daizhang-backend/src/main/java/com/company/daizhang/module/subject/entity/Subject.java
package com.company.daizhang.module.subject.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.company.daizhang.common.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 科目实体
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("acc_subject")
public class Subject extends BaseEntity {
    
    private Long accountSetId;
    
    private String code;
    
    private String name;
    
    private String category;
    
    private Long parentId;
    
    private Integer level;
    
    private Integer balanceDirection;
    
    private Integer isAuxiliary;
    
    private Integer isCash;
    
    private Integer isBank;
    
    private Integer isCurrent;
    
    private Integer status;
}
